packages/react-router-dom/modules/Link.js

```diff
@@ -52,9 +57,24 @@ class Link extends React.Component {
     }
   }
 
-  render() {
-    const { replace, to, ...props } = this.props // eslint-disable-line no-unused-vars
+  absolute() {
+    const { to } = this.props
+    const { match } = this.context.router
+    const base = (match && match.url) ? match.url : ''
+    return resolveLocation(to, base)
+  }
+
+  componentWillMount() {
+    this.unlisten = this.context.router.listen(() => { this.forceUpdate() })
```

Comment:
@pshrmn yeah, the original v4 code didn't do all this subscribe crap. But the React team insists that accessing data on context (in this case location) is not reliable and have asked us to use subscriptions, which is a real bummer.